packages/web/src/common/store/notifications/fetchNotificationsSaga.ts

```diff
@@ -0,0 +1,62 @@
+import {
+  FetchNotifications,
+  getErrorMessage,
+  notificationsActions,
+  notificationsSelectors
+} from '@audius/common'
+import moment from 'moment'
+import { call, put, select, takeEvery } from 'typed-redux-saga'
+
+import { NOTIFICATION_LIMIT_DEFAULT } from './constants'
+import { fetchNotifications } from './fetchNotifications'
+import { parseAndProcessNotifications } from './parseAndProcessNotifications'
+
+const { getLastNotification } = notificationsSelectors
+const {
+  fetchNotificationsRequested,
+  fetchNotificationsFailed,
+  fetchNotificationSucceeded
+} = notificationsActions
+
+export function* watchFetchNotifications() {
+  yield* takeEvery(
+    notificationsActions.FETCH_NOTIFICATIONS,
+    fetchNotificationsSaga
+  )
+}
+
+/**
+ * Fetch notifications, used by notification pagination
+ * This is the function used to fetch more notifcations after the initial load in getNotifications
+ */
+function* fetchNotificationsSaga(action: FetchNotifications) {
+  yield* put(fetchNotificationsRequested())
+  const limit = action.limit || NOTIFICATION_LIMIT_DEFAULT
+  const lastNotification = yield* select(getLastNotification)
+  const timeOffset = lastNotification
+    ? lastNotification.timestamp
+    : moment().toISOString()
+
+  const notificationsResponse = yield* call(fetchNotifications, {
+    limit,
+    timeOffset,
+    groupIdOffset: lastNotification?.groupId
+  })
+
+  if ('error' in notificationsResponse) {
+    yield* put(
+      fetchNotificationsFailed(getErrorMessage(notificationsResponse.error))
```

Comment:
Should we log this as a metric? I guess it's likely due to network errors, so maybe not that effective.

Reply:
so just to double check, youre advocating for or against a metric?

Reply:
Ah, sorry. I think it's probably more effort than it's worth.

Reply:
we do have an errorSaga that captures the failure, and if we pass "shouldReport" then itll report to sentry so easy to add either way!

Reply:
🤷 I mean it would be good to know if notifications are failing, as it feels like a relatively quiet failure mode. If it's easy, let's do it. Should also remember to add the alerting for it then.